packages/client/logic/nav.ts

```diff
@@ -64,206 +27,41 @@ export const queryClicks = computed({
   },
 })
 
-export const rawTree = computed(() => slides.value
-  .filter((route: SlideRoute) => route.meta?.slide?.title)
-  .reduce((acc: TocItem[], route: SlideRoute) => {
-    addToTree(acc, route)
-    return acc
-  }, []))
-export const treeWithActiveStatuses = computed(() => getTreeWithActiveStatuses(rawTree.value, currentSlideRoute.value))
-export const tree = computed(() => filterTree(treeWithActiveStatuses.value))
-
-export const transition = computed(() => getCurrentTransition(navDirection.value, currentSlideRoute.value, prevRoute.value))
-
-watch(currentSlideRoute, (next, prev) => {
-  navDirection.value = next.no - prev.no
-})
-
-watch([total, route], async () => {
-  if (hasPrimarySlide.value && !getSlide(route.value.params.no as string)) {
+export const {
+  slides,
+  total,
+  currentPath,
+  currentSlideNo,
+  currentPage,
+  currentLayout,
+  currentTransition,
+  clicksDirection,
+  nextRoute,
+  prevRoute,
+  clicks,
+  clicksTotal,
+  hasNext,
+  hasPrev,
+  tocTree,
+  navDirection,
+  openInEditor,
+  next,
+  prev,
+  go,
+  goLast,
+  goFirst,
+  nextSlide,
+  prevSlide,
+} = useNavBase(
+  currentSlideRoute,
+  clicksContext,
+  queryClicks,
+  router,
+)
+
+watch([total, currentSlideNo], async () => {
+  if (hasPrimarySlide.value && !getSlide(currentSlideNo.value)) {
```

Comment:
```suggestion
watch([total, route], async () => {
  if (hasPrimarySlide.value && !getSlide(route.value.params.no as string)) {
```

`currentSlideNo` is computed, and is always in the range of slides.